staging/src/k8s.io/apiserver/pkg/endpoints/audit_test.go

```diff
@@ -140,8 +149,8 @@ func TestAudit(t *testing.T) {
 			200,
 			2,
 			[]eventCheck{
-				noRequestBody(0),
-				responseBodyMatches(0, `{.*"name":"c".*}`),
+				noRequestBody(1),
```

Comment:
So as the test stands when you get save the events to the sink list you are just saving the pointer, that event goes on to be built up in the handler. The result is you have two pointers to the same final event. These tests were passing because the first event was changed to the `ResponseComplete` which is what these tests intended on targeting.

Reply:
Got it, thanks! Should we add a check that verifies the event is the expected stage?

Reply:
yep good idea

Reply:
ok updated, went the composable route